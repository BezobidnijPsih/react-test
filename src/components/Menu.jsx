import React, {Component} from 'react';
import {ListGroup, ListGroupItem} from 'react-bootstrap';
import {routes} from './../Routes';
import {Link} from 'react-router-dom';
import styled from 'styled-components';

class Menu extends Component {
  render() {
    const MenuLink = styled(Link)`
      &:hover {
        text-decoration: none;
      }
    `;
    return (
      <ListGroup>
        {routes.filter(route => route.mainMenuLabel).map(route => (
          <ListGroupItem key={route.key}>
            <MenuLink to={route.path}>{route.mainMenuLabel}</MenuLink>
          </ListGroupItem>
        ))}
      </ListGroup>
    );
  }
}

export default Menu;
