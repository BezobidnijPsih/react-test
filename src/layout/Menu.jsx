import React, {Component} from 'react';
import {ListGroup, ListGroupItem} from 'react-bootstrap';
import {routes} from '../app/Routes';
import styled from 'styled-components';
import {LinkContainer} from 'react-router-bootstrap';

class Menu extends Component {
  render() {
    return (
      <ListGroup>
        {routes.filter(route => route.mainMenuLabel).map(route => (
          <LinkContainer
            exact
            to={route.path}
            key={route.key}
            className="list-group-item">
            <a>{route.mainMenuLabel}</a>
          </LinkContainer>
        ))}
      </ListGroup>
    );
  }
}

export default Menu;
