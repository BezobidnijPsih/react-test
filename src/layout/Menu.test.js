import React from 'react';
import Menu from './Menu';
import toJson from 'enzyme-to-json';

describe('Menu', () => {
  test('should render correctly', () => {
    const menu = shallow(<Menu />);
    expect(toJson(menu)).toMatchSnapshot();
  });
});
