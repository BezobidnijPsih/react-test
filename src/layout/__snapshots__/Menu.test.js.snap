// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render correctly 1`] = `
<ListGroup
  bsClass="list-group"
>
  <ListGroupItem
    bsClass="list-group-item"
    key="cars"
    listItem={false}
  >
    <Styled(Link)
      to="/"
    >
      Cars
    </Styled(Link)>
  </ListGroupItem>
  <ListGroupItem
    bsClass="list-group-item"
    key="races"
    listItem={false}
  >
    <Styled(Link)
      to="/races"
    >
      Races
    </Styled(Link)>
  </ListGroupItem>
  <ListGroupItem
    bsClass="list-group-item"
    key="tracks"
    listItem={false}
  >
    <Styled(Link)
      to="/tracks"
    >
      Tracks
    </Styled(Link)>
  </ListGroupItem>
  <ListGroupItem
    bsClass="list-group-item"
    key="users"
    listItem={false}
  >
    <Styled(Link)
      to="/users"
    >
      Users
    </Styled(Link)>
  </ListGroupItem>
</ListGroup>
`;
